server/app/rest_api/rest.py

```diff
@@ -224,13 +223,96 @@ def get(self):
                 "examples": {
                     "application/json": {
                         "names": [
-                            'tissue_type', 'sex', 'num_reads', 'clusters'
+                            "tissue_type", "sex", "num_reads", "clusters"
                         ],
                         "data": [
-                            [0, 'lung', 'F', 39844, 99],
-                            [1, 'heart', 'M', 83, 1],
-                            [49, 'spleen', None, 2, "unknown cluster"],
+                            [0, "lung", "F", 39844, 99],
+                            [1, "heart", "M", 83, 1],
+                            [49, "spleen", None, 2, "unknown cluster"],
+
+                        ]
+                    }
+
+                }
+            }
+        }
+    })
+    def put(self):
+        fields = request.args.getlist("annotation-name", None)
+        df = current_app.data.filter_dataframe(request.get_json()["filter"])
+        try:
+            annotation_response = current_app.data.annotation(df, "obs", fields)
+        except KeyError:
+            return make_response(f"Error bad key in {fields}", 404)
+        return make_response(jsonify(annotation_response))
+
 
+class AnnotationsVarAPI(Resource):
+    @swagger.doc({
+        "summary": "Fetch annotations (metadata) for all variables.",
+        "tags": ["annotations"],
+        "parameters": [{
+            "in": "query",
+            "name": "annotation-name",
+            "type": "string",
+            "description": "list of 1 or more annotation names"
+        }],
+        "responses": {
+            "200": {
+                "description": "annotations",
+                "examples": {
+                    "application/json": {
+                        "names": [
+                            "name", "category"
+                        ],
+                        "data": [
+                            [0, "ATAD3C", 1],
+                            [1, "RER1", None],
+                            [49, "S100B", 6]
+                        ]
+                    }
+
+                }
+            }
+        }
+    })
+    def get(self):
```

Comment:
is this not the same as the put request?  i understand this is because some requests might exceed the 2k or so limit for GET requests, but maybe you can have them call the same code.